### PR TITLE
Enhanced Injector API

### PR DESCRIPTION
**Pull request by mayurarora0903**

### CODE REVIEW

**Comment by mayurarora0903** [comment]
This way we ( consumers ) would not need to include an additional script ( referred to as the injector script till now ). This enables easier integration and reduces the overall surface area of the plugin.

File: lib/sw-utils.js [hunk 1]
```diff
@@ -1,27 +1,79 @@
 'use strict';
 
-function injectServiceWorker(updateCallback) {
+var createFrame = function createFrame(id, innerHTML, classList) {
```
Comment (shubh54) [inline lib/sw-utils.js]: either use function createFrame or use var createFrame = function() {}

File: lib/sw-utils.js [hunk 2]
```diff
@@ -1,27 +1,79 @@
...
+var hideFrame = function hideFrame(id) {
+    return document.getElementById(id).classList.add('hidden-frame');
+};
+
+var showFrame = function showFrame(id) {
+    return document.getElementById(id).classList.remove('hidden-frame');
+};
+
+function updateExperience(sw) {
+    var refreshing = void 0;
```
Comment (shubh54) [inline lib/sw-utils.js]: is void 0 needed ?

File: lib/sw-utils.js [hunk 3]
```diff
@@ -1,27 +1,79 @@
...
+    document.getElementById('ignore-update').addEventListener('click', function () {
+        return hideFrame('app-update-frame');
+    });
+    navigator.serviceWorker.addEventListener('controllerchange', function () {
```
Comment (shubh54) [inline lib/sw-utils.js]: add conditional if serviceWorker in navigator for all such operations.

File: src/sw-utils.js [hunk 4]
```diff
@@ -13,32 +13,74 @@
...
+};
+
+const hideFrame = id => document.getElementById(id).classList.add('hidden-frame');
+
+const showFrame = id => document.getElementById(id).classList.remove('hidden-frame');
+
+function updateExperience(sw) {
+    let refreshing;
+    document.getElementById('ignore-update').addEventListener('click', () => hideFrame('app-update-frame'));
+    navigator.serviceWorker.addEventListener(
```
Comment (shubh54) [inline src/sw-utils.js:34]: add conditional serviceWorker in navigator

File: src/sw-utils.js [hunk 5]
```diff
@@ -13,32 +13,74 @@
...
+        });
+    });
+}
+
+function injectServiceWorker({ withUpdate = false, cb } = {}) {
     if ('serviceWorker' in navigator) {
         navigator.serviceWorker.register('/service-worker.js')
             .then((reg) => {
+                if (withUpdate) {
+                    const appUpdateTemplate = "<div id='update-txt'><img src='https://retail.flixcart.com/www/fk-retail-vpp/icon-star.png' /><span id='message'>Catch up with the latest version.</span><a id='update-app'>Update</a></span><a id='ignore-update'>X</a></div>";
```
Comment (shubh54) [inline src/sw-utils.js]: why not separate out these constants to separate file will make this function cleaner

**Review by shubh54 — APPROVED** [review]

**Pull request merged**